src/util/misc/matrix.ts

```diff
@@ -84,6 +82,46 @@ export const multiplyTransformMatrices = (
     is2x2 ? 0 : a[1] * b[4] + a[3] * b[5] + a[5],
   ] as TMat2D;
 
+/**
+ * Multiplies {@link matrices} such that a matrix defines the plane for the rest of the matrices **after** it
+ *
+ * `multiplyTransformMatrixChain([A, B, C, D])` is equivalent to `A(B(C(D)))`
+ *
+ * @param matrices an array of matrices
+ * @param [is2x2] flag to multiply matrices as 2x2 matrices
+ * @returns the multiplication product
+ */
+export const multiplyTransformMatricesFromEnd = (
+  matrices: (TMat2D | undefined | null | false)[],
+  is2x2?: boolean
+) =>
+  matrices.reduceRight(
+    (product: TMat2D, curr) =>
+      curr ? multiplyTransformMatrices(curr, product, is2x2) : product,
+    iMatrix
+  );
+
+/**
+ * Multiplies {@link matrices} such that a matrix defines the plane for the rest of the matrices **before** it
+ *
+ * `multiplyTransformMatrixChain([A, B, C, D])` is equivalent to `D(C(B(A)))`
+ *
+ * **Note**: this form of multiplication is considered as reversed mathematically
+ *
+ * @param matrices an array of matrices
+ * @param [is2x2] flag to multiply matrices as 2x2 matrices
+ * @returns the multiplication product
+ */
+export const multiplyTransformMatricesFromStart = (
```

Comment:
maybe this is a bad idea because it is equal to `multiplyTransformMatricesFromEnd(matrices.reverse())`